app/models/notification.rb

```diff
@@ -71,4 +72,9 @@ def notification_has_at_least_one_recipient
     end
   end
 
+  def expired?
+    Notification.where(id_code: id_code)
+      .select { |notification| notification.end_time > 6.months.ago }.empty?
```

Comment:
Ah, now this one makes more sense, because it has the `.empty?` on the end.

Reply:
Same here - if the notification is already persisted, I think this search needs to exclude `self.id`, or the record may appear to conflict with itself during an update.